textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -25,48 +20,49 @@
 import edu.uci.ics.textdb.api.common.Attribute;
 import edu.uci.ics.textdb.api.common.FieldType;
 import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.IPredicate;
 import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
 import edu.uci.ics.textdb.api.storage.IDataReader;
-import edu.uci.ics.textdb.api.storage.IDataStore;
 import edu.uci.ics.textdb.common.exception.DataFlowException;
 import edu.uci.ics.textdb.common.exception.ErrorMessages;
 import edu.uci.ics.textdb.common.field.DataTuple;
 import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.storage.DataReaderPredicate;
 
 /**
  * @author sandeepreddy602
  *
  */
-public class LuceneDataReader implements IDataReader{
-    
-    private IDataStore dataStore;
-    private int cursor = -1;
+public class DataReader implements IDataReader{
+
+    private int cursor = CLOSED;
     private IndexSearcher indexSearcher;
     private ScoreDoc[] scoreDocs;
     private IndexReader indexReader;
-    private Query query;
-    
-    public LuceneDataReader(IDataStore dataStore, Query query) {
-        this.dataStore = dataStore;
-        this.query = query;
+    private DataReaderPredicate dataReaderPredicate;
+
+    public DataReader(IPredicate dataReaderPredicate) {
+        this.dataReaderPredicate = (DataReaderPredicate)dataReaderPredicate;
     }
     
     @Override
     public void open() throws DataFlowException {
 
 
         try {
-            Directory directory = FSDirectory.open(Paths.get(dataStore.getDataDirectory()));
+            String dataDirectory = dataReaderPredicate.getDataStore().getDataDirectory();
+            Directory directory = FSDirectory.open(Paths.get(dataDirectory));
             indexReader = DirectoryReader.open(directory);
                 		
             indexSearcher = new IndexSearcher(indexReader);

```

Comment:
"indexSearcher" is from Lucene. Let's rename it to "luceneIndexSearcher" to make it more clear and different from our classes. In general, it will be better to add a "lucene" prefix to those Lucene-related variables.